tldr/diagnostics.py

```
@@ -936,6 +1011,27 @@ def get_diagnostics(
             except subprocess.TimeoutExpired:
                 pass
 
+    elif lang == "r":
+        # No type checker for R (dynamically typed)
+        # Run lintr for linting
+        if include_lint and shutil.which("Rscript"):
+            try:
+                # lintr is an R package, run via Rscript
+                result = subprocess.run(
```

Ruff (0.14.13)

`subprocess` call: check for execution of untrusted input

(S603)